src/main.py

```diff
@@ -53,6 +53,9 @@ def __init__(self):
         # initialize SLMDemo
         self.SLM = SLMDemo()
         self.devices['SLM'] = self.SLM
+        
+        self.SLM.slm_worker.start()
+        self.SLM.slm_worker.slmParamsSignal.connect(self.SLM.handle_slm_params)
```

Comment:
This seems to be machinery internal to the SLM device. Could you change the code so that this line is not required and is instead handled in the constructor of the SLM device?